### DISCUSSION TITLE
PIR on Sonoff Dual R3

### DISCUSSION THREAD

**Discussion by DSew20** [arendst/Tasmota#13956] [Q&A] [answered]
Hello.  I have been trying (without success) to configure a Sonoff Dual R3 (with Tasmota 10) to operate via a connected PIR.

So, the Sonoff Dual R3 has 2 relays and 2 switches.  One stitch has a 230v PIR, the second in currently not used.  The relays are connected to 2 x floodlights.  I am using Home Assistant, and ideally I would like the PIR to:

1) switch on both the floodlights for 2mins if there is movement after dark.  Ideally I would like this to be a rule so not via Home Assistant
2) report any movement to Home Assistant at any time of the day (via MQTT or directly as a sensor)

To add, the PIR is set to detect at all lux levels and the in-built timer is set to minimum, which is around 15 seconds.  I guess I am interested in the leading edge upon movement being detected.

I have tried playing with SwitchMode, SwitchTopic and SetOption115 but just can't seem to get this to work.  I am guessing others have solved for this, so was looking for any pointers / tips on how to achieve such a setup.

Thanks

**Comment by ascillato** [comment]
Hi, please try the examples posted in the docs. You need to use rules. In the rules cookbook there is an example similar to what you need.

Also, in older issues, you can find more examples.

**Comment by rapi3** [comment, marked as answer]
Have a look here how to set your PIR for Tasmota and HA.
https://blakadder.com/pir-in-tasmota-integration/

The rest you can find in doc example and experiment.

Here I am using a Sonoff R1 to control the light at stairs from few sensors and cctv and to turn it ON only at dark with timers:

`rule1
ON Power1#Boot DO var1 1 ENDON
ON mqtt#connected DO Backlog Subscribe w13cctv, s/W13/CCTV; Subscribe w03pir, s/W03/PIR; Subscribe w07pir, s/W07/PIR; Subscribe rpi3cctv, LLC/cctv/rpi1 ENDON
ON event#w13cctv=ON DO IF (%var1%!=0) Power1 1 ENDIF ENDON
ON event#w03pir=ON DO IF (%var1%!=0) Power1 1 ENDIF ENDON
ON event#w07pir=ON DO IF (%var1%!=0) Power1 1 ENDIF ENDON
ON event#rpi3cctv=Alarm DO IF (%var1%!=0) Power1 1 ENDIF ENDON

rule2 ON clock#Timer=1 DO var1 0 ENDON ON clock#Timer=2 DO var1 1 ENDON`
![Screenshot_2021-12-08_14-59-51](https://user-images.githubusercontent.com/18531150/145221206-04af8ca1-48a3-4249-9cbe-27a2f0d266ef.png)
![Screenshot_2021-12-08_15-00-04](https://user-images.githubusercontent.com/18531150/145221222-e7f07407-76a1-45c2-9551-34a57d1c85d4.png)

**Reply by DSew20** [reply]
Thank you.  This really helped.

**Comment by DSew20** [comment]
Thank you.  This is helpful.

The PIR example from the cookbook looks good:

`Rule1
   ON Switch1#state=1 DO Backlog event checksunrise=%time%; event checksunset=%time% ENDON
   ON event#checksunrise<%sunrise% DO Power1 1 ENDON
   ON event#checksunset>%sunset% DO Power1 1 ENDON` 

I assume I can also add the following to the Backlog to send an MQTT on movement `Publish stat/%topic%/PIR1 ON`

But, to get this working does one also need to set the detach the switch from the relay?  And / or set SwitchTopic to 1 so that MQTT messages are also sent?

**Comment by DSew20** [comment]
So, this seems to work if anyone wants to do the same:

PIR connected across Switch1.  Relay1 connected to lights.

The PIR sends an MQTT whenever there is movement.  But, only turns the light on for 1min between sunset and sunrise.  Hope this helps others

`SwitchMode1 13
PulseTime 60

Rule1
  ON Switch1#state=1 DO Backlog event checksunrise=%time%; event checksunset=%time%;  publish stat/%topic%/MOTION ON ENDON
  ON event#checksunrise<%sunrise% DO Power1 1 ENDON
  ON event#checksunset>%sunset% DO Power1 1 ENDON`

Then `rule1 1` to activate.
